clients/src/test/java/org/apache/kafka/clients/MockClient.java

```diff
@@ -263,8 +263,11 @@ private synchronized void maybeAwaitWakeup() {
             if (remainingBlockingWakeups <= 0)
                 return;
 
-            while (numBlockingWakeups == remainingBlockingWakeups)
-                wait();
+            TestUtils.waitForCondition(() -> {
+                if (numBlockingWakeups == remainingBlockingWakeups)
+                    MockClient.this.wait(500);
```

Comment:
```TestUtils.waitForCondition``` already call ```sleep``` so is this ```wait``` necessary?

Reply:
I was thinking the wait would allow for faster notification.